src/route.shared.ts

```diff
@@ -281,3 +279,112 @@ function isErrorPage(pageId: string): boolean {
+    .filter(route => isCallable(route.pageRoute))
+
+  const routeFunctionResults : RouteFunctionResult[] = functionalRoutes
+    .map(route => {
+      assert(route.pageRouteFile);
+      const routeFunctionResult : (string|RouteFunctionMatch) = resolveRouteFunction(route.pageRoute, url, contextProps, route.pageRouteFile);
+      
+      return { ...route, pageRoute: routeFunctionResult };
+    });
+
+  return routeFunctionResults
+    .sort(sortRoutes)
+}
+
+function getRouteStrings(routes: PageRoute[], pageIds: PageId[]) {
+  const { sortRoutes } = getCustomRouter();
+
+  const fsRouteStrings : PageRoute[] = pageIds
+    .filter(pageId => !routes.some(route => route.id === pageId) && !isErrorPage(pageId))
+    .map(id => ({ pageRoute: getFilesystemRoute(id, pageIds), id }));
+
+  const routeStrings = Object.values(routes)
+    .filter(route => !isCallable(route.pageRoute));
+
+  return [
+    ...fsRouteStrings,
+    ...routeStrings
+  ].sort(sortRoutes)
+}
+
+const getMatchVal = (route: PageRoute): number => 
+  typeof route.pageRoute === 'string' 
+  ? route.pageRoute.length
+  : route.pageRoute.constructor === Object && !isCallable(route.pageRoute)
+    ? typeof route.pageRoute.matchValue === 'number'
+      ? route.pageRoute.matchValue
+      : route.pageRoute.matchValue
+        ? 1
+        : 0
+    : 0;
+
+function defaultSortRoutes(a: PageRoute, b: PageRoute): number {
+  return getMatchVal(b) - getMatchVal(a);
+}
+
+/* pathToRegexp route handling. These should be moved out (possibly to a separate package). */
+
+async function matchPathToRegexpRoutes(
+  routes: PageRoute[],
+  url: string
+): Promise<null | undefined | RouteMatch> {
+
+  for (var ii = 0; ii < routes.length; ++ii) {
+    const route = routes[ii];
+    const { pageRoute, id: pageId } = route;
+
+    // Route with `.page.route.js` defined route string
+    if (typeof pageRoute === 'string') {
+      const { matchValue, routeParams } = routeWith_pathToRegexp(url, pageRoute)
+      return { pageId, routeParams }
+    }
+
+    // Route with `.page.route.js` defined route function
+    if (pageRoute.constructor === Object) {
+      const { matchValue, routeParams } = pageRoute as RouteFunctionMatch;
+      return { pageId, routeParams }
+    }
+  }
+  return null;
+}
+
+declare global {
+  namespace NodeJS {
+    interface Global {
+      __vite_ssr_plugin_custom_router: RoutingHandler
+    }
+  }
+}
+
+declare global {
+  interface Window {
+    __vite_ssr_plugin_custom_router: RoutingHandler
+  }
+}
+
+function getCustomRouter() {
+  if( typeof window !== "undefined") {
+    // Browser
+    return window.__vite_ssr_plugin_custom_router
+  } else {
+    // Node.js; `global.customRouter` has been set somewhere during the `createPageRender()` call.
+    return global.__vite_ssr_plugin_custom_router
+  }
+}
+// @TODO eventually the pathToRegexp implementation should be removed from this file entirely so as not to bloat the bundle when it is unused
```

Comment:
> so as not to bloat the bundle when it is unused

FYI, I'm afraid this is not possible as things currently are: bundling for the browser is done statically whereas replacing pathToRegexp with a custom router is done at runtime. (But a solution would be to not implicity include pathToRegexp: the user would have to always set `customRouter` (and we would rename `customRouter` to `router`).)

But yes we should still remove it from this file for code readability once the PR is merged.